compiler/rustc_mir/src/transform/check_consts/ops.rs

```diff
@@ -10,31 +10,34 @@ use rustc_span::{Span, Symbol};
 
 use super::ConstCx;
 
-/// Emits an error if `op` is not allowed in the given const context.
-pub fn non_const<O: NonConstOp>(ccx: &ConstCx<'_, '_>, op: O, span: Span) {
+/// Emits an error and returns `true` if `op` is not allowed in the given const context.
+pub fn non_const<O: NonConstOp>(ccx: &ConstCx<'_, '_>, op: O, span: Span) -> bool {
```

Comment:
Yeah, it's possible that this helper function should folded back into the const-checker, since the return value and (unfortunately) the `unstable_in_stable` bit are unused in the post-elaboration live drop checker.